libraries/AC_Fence/AC_Fence.h

```diff
@@ -124,10 +126,10 @@ class AC_Fence
     uint8_t get_action() const { return _action.get(); }
 
     /// get_safe_alt - returns maximum safe altitude (i.e. alt_max - margin)
-    float get_safe_alt_max() const { return _alt_max - _margin; }
+    float get_safe_alt_max() const { return get_alt_max_relhome() - _margin; }
 
     /// get_safe_alt_min - returns the minimum safe altitude (i.e. alt_min + margin)
-    float get_safe_alt_min() const { return _alt_min + _margin; }
+    float get_safe_alt_min() const { return get_alt_min_relhome() + _margin; }
```

Comment:
```suggestion
    float get_safe_alt_min(Location::AltFrame _frame) const { return get_alt_min_relhome() + _margin; }
```

... and again, this probably has to be boolean.

Reply:
Updated this comment as the implementation would need to change